### PR TITLE
Feature/aggregation per iteration

### PR DESCRIPTION
**Pull request by daniel-sol**

### CODE REVIEW

File: src/sumo/table_aggregation/dispatch.py [hunk 1]
```diff
@@ -7,7 +7,9 @@
 from httpx import HTTPStatusError
 
 
-def query_for_it_name_and_tags(sumo: SumoClient, case_uuid: str, pit):
+def query_for_names_and_tags(
+    sumo: SumoClient, case_uuid: str, iter_name: str = "0", pit=None
+):
```
Comment (equinor-ruaj) [inline src/sumo/table_aggregation/dispatch.py:12]: Is it correct that we don't use the `pit` anymore?
If so, should remove it from all relevant places.

File: src/sumo/table_aggregation/dispatch.py [hunk 2]
```diff
@@ -162,50 +116,48 @@ def generate_dispatch_info(uuid, env, token=None, pit=None, seg_length=250):
...
-                dispatch_combination["tag_name"] = tag_name
-                for col_segment in list_of_list_segments(
-                    sumo,
-                    uuid,
+    dispatch_combination = {}
+    dispatch_combination["uuid"] = uuid
+    # dispatch_combination["iter_name"] = iteration_name
+    # for iter_name, it_tables in it_name_and_tag.items():
+    #     logger.debug(iter_name)
+    #     logger.debug(it_tables)
```
Comment (equinor-ruaj) [inline src/sumo/table_aggregation/dispatch.py:128]: remove comment?

File: src/sumo/table_aggregation/utilities.py [hunk 3]
```diff
@@ -108,7 +108,7 @@ def init_logging(name: str) -> logging.Logger:
     returns (logging.Logger): an initialized logger
     """
     logger = logging.getLogger(name)
-    logger.addHandler(logging.NullHandler())
+    # logger.addHandler(logging.NullHandler())
```
Comment (equinor-ruaj) [inline src/sumo/table_aggregation/utilities.py:111]: remove?

File: src/sumo/table_aggregation/utilities.py [hunk 4]
```diff
@@ -469,6 +410,39 @@ def uuid_from_string(string: str) -> str:
...
+        try:
+            table = pq.read_table(file_path, columns=available_columns)
+            logger.warning("Got %s columns less than asked for", len_asked_for - len_retrieved)
+        except pa.lib.ArrowInvalid:
+            table = pa.table([])
+            logger.error("file %s is empty", file_path)
+    except pa.lib.ArrowInvalid:
+        table = pa.table([])
+        logger.error("Table with name %s is completely empty", file_path)
+    return table
```
Comment (equinor-ruaj) [inline src/sumo/table_aggregation/utilities.py:444]: I guess this also works for "complete" tables so maybe better to just name it `read_table` ?

File: src/sumo/table_aggregation/utilities.py [hunk 5]
```diff
@@ -488,8 +463,15 @@ def get_object(object_id: str, cols_to_read: list, sumo: SumoClient) -> pa.Table
 
         table = blob_to_table(BytesIO(blob.content))
         pq.write_table(table, file_path)
+        logger.debug("Written object to file %s", file_path)
+    try:
+        table = pq.read_table(file_path, columns=list(cols_to_read))
+        logger.debug("Table is read as should be!")
+    except (pa.lib.ArrowInvalid, KeyError):
+
+        table = read_incomplete_table(file_path, cols_to_read)
```
Comment (equinor-ruaj) [inline src/sumo/table_aggregation/utilities.py:472]: Do we need try/except here or will `read_incomplete_table` work in both cases?

File: src/sumo/table_aggregation/utilities.py [hunk 6]
```diff
@@ -725,21 +707,25 @@ def reconstruct_table(
...
             logger.info("Real: %s, missing these columns %s", real_nr, missing)
+
         for miss in missing:
             real_table = real_table.add_column(0, miss, pa.array([None] * rows))
         logger.debug("Table created %s", type(real_table))
 
         # Sort to ensure that table has cols in same order even
         # when missing cols occur
-        real_table = real_table.select(sorted(real_table.column_names))
+        # real_table = real_table.select(sorted(real_table.column_names))
```
Comment (equinor-ruaj) [inline src/sumo/table_aggregation/utilities.py:725]: remove commented code?

File: src/sumo/table_aggregation/utilities.py [hunk 7]
```diff
@@ -880,23 +873,26 @@ def prepare_object_launch(meta: dict, table, name, operation):
...
         parent
         + f"--{name}--{tag}--{operation}--"
         + full_meta["fmu"]["iteration"]["name"]
     )
+    print(f"Full meta: {full_meta}", flush=True)
     full_meta["file"]["checksum_md5"] = md5
+    # full_meta["fmu"]["aggregation"] = {
+    #     "id": uuid_from_string(md5),
+    #     "operation": operation,
+    # }
```
Comment (equinor-ruaj) [inline src/sumo/table_aggregation/utilities.py:886]: remove `print` and commented code?

File: src/sumo/table_aggregation/dispatch.py (hunk 1 above)
Comment (daniel-sol) [reply src/sumo/table_aggregation/dispatch.py]: Yes, removed

File: src/sumo/table_aggregation/dispatch.py (hunk 2 above)
Comment (daniel-sol) [reply src/sumo/table_aggregation/dispatch.py:121]: Yup, removed

File: src/sumo/table_aggregation/utilities.py (hunk 3 above)
Comment (daniel-sol) [reply src/sumo/table_aggregation/utilities.py]: Yes. done!

File: src/sumo/table_aggregation/utilities.py (hunk 4 above)
Comment (daniel-sol) [reply src/sumo/table_aggregation/utilities.py:443]: renamed to read_available_columns

File: src/sumo/table_aggregation/utilities.py (hunk 6 above)
Comment (daniel-sol) [reply src/sumo/table_aggregation/utilities.py:724]: Yes, done.

File: src/sumo/table_aggregation/utilities.py (hunk 5 above)
Comment (daniel-sol) [reply src/sumo/table_aggregation/utilities.py]: Keeping it for now, will test further

File: src/sumo/table_aggregation/utilities.py (hunk 7 above)
Comment (daniel-sol) [reply src/sumo/table_aggregation/utilities.py]: Yes!

**Review by equinor-ruaj — APPROVED** [review]
lgtm

**Pull request merged**